markdown/extensions/toc.py

```diff
@@ -16,7 +16,7 @@
 from . import Extension
 from ..treeprocessors import Treeprocessor
 from ..util import code_escape, parseBoolValue, AMP_SUBSTITUTE, HTML_PLACEHOLDER_RE, AtomicString
-from ..postprocessors import UnescapePostprocessor
+from ..treeprocessors import UnescapeTreeprocessor
```

Comment:
I excepted to not need to explicitly call the unescaping code from the TOC extension. After all, the TOC extension updates the Tree prior to the `UnescapeTreeprocessor` being run. However, when I remove the explicit calls to escape, the tests which specifically test this code fail. I haven't worked out why.

Reply:
I figured out the reason. The text of a TOC item is passed to `slugify` for the `id`. However, it needs to be unescaped beforehand. Otherwise, `slugify` mucks up the placeholder so that the new Treeprocessor doesn't recognize it and fails to make the swap in the `id`. However, the text in the body of the TOC item does not need unescaped within the TOC extension as it will be handled by the new Treeprocessor.